scripts/ai/AIDriveStrategyUnloadCombine.lua

```diff
@@ -243,12 +291,15 @@ function AIDriveStrategyUnloadCombine:getDriveData(dt, vX, vY, vZ)
     end
 
     -- make sure if we have a combine we stay registered
-    if self.combineToUnload then
-        self.combineToUnload:getCpDriveStrategy():registerUnloader(self)
+    if self.combineToUnload and self.combineToUnload:getIsCpActive() then
+        if not self.combineToUnload:getCpDriveStrategy():registerUnloader(self) then 
+            --- Resets the unload target, when the unload target type changed.
```

Comment:
Who changes it? Where? Wouldn't it be more straightforward to reset it where it is being changed?